### DISCUSSION TITLE
HTTPS request for meeting CSV returns 400

### DISCUSSION THREAD

**Discussion by landolfi-us** [code4recovery/12-step-meeting-list#1339] [Q&A] [answered]
Hi,

I'm invoking the TSML CSV export URL from a Powershell script  that also automatically downloads a Formidable export CSV, then compares the two to determine differences. The TSML download URL is a copy of the URL on the TSML export button and has the form "https://xxxxx.org/wp-admin/admin-ajax.php?action=csv". This URL recently used to work  with Powershell invoke-Webrequest, but for some reason it now returns 400, which seems to be coming from admin-ajax.php. The same URL works fine in a browser provided I'm logged in as admin. I thought maybe I needed to pass the site username and password credentials as well as the HTTPS URL, but that still gets a 400 response for some reason. Any ideas what else I need to do to get the non-browser download to work?

**Comment by joshreisner** [comment, marked as answer]
hey, we recently revoked access to the CSV for non-logged in users, because it was determined to be a security breach :(

for scripting data access we recommend using the data feed

**Reply by landolfi-us** [reply]
Thanks, that explains it. Is it possible to revert that change somehow if I'm willing to accept the risk? Or is there a way to become logged in as far as Wordpress is concerned outside of a browser?

**Reply by joshreisner** [reply]
yeah. we can make this easier for you in a future release but for now you could take the code block here: https://github.com/code4recovery/12-step-meeting-list/blob/main/includes/ajax.php#L185-L244

and copy it to your theme's functions.php and change `wp_ajax_csv` in the first line of your version to `wp_ajax_nopriv_csv`

**Reply by landolfi-us** [reply]
Thanks, Josh. On your suggestion I just switched to JSON, it's 2 additional lines of Powershell code to convert JSON to CSV. But thanks for the snippet, it helps me gain insight into how WP handles this kind of request.

**Reply by landolfi-us** [reply]
Actually it turns out Powershell doesn't handle optional JSON fields very gracefully, so I created the code snippet described above and there's definitely progress, but now I get a download containing this: "You do not have sufficient permissions to access this page (<code>edit_posts</code>).". I tried including credentials with the request but no joy. Is there something else I need to do?

**Reply by joshreisner** [reply]
yah take out `tsml_require_meetings_permission();` from your version - sorry should have called that out earlier

**Reply by landolfi-us** [reply]
Thanks!
